modules/api/pkg/resources/machine/common.go

```diff
@@ -273,30 +274,48 @@ func getVSphereProviderSpec(c *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc
 	return EncodeAsRawExtension(config)
 }
 
-func getBaremetalProviderSpec(cluster *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*runtime.RawExtension, error) {
-	// Check if Tinkerbell configuration is provided
+func GetBaremetalProviderConfig(cluster *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*baremetal.RawConfig, error) {
+	// Ensure Tinkerbell configuration is provided
 	if nodeSpec.Cloud.Baremetal.Tinkerbell == nil {
-		return nil, errors.New("tinkerbell provisioner is required")
+		return nil, errors.New("tinkerbell provisioner is required in baremetal configuration")
 	}
 
-	// Create TinkerbellNodeSpec from nodeSpec
-	tinkerbellSpec := apiv1.TinkerbellNodeSpec{
-		OsImageUrl:  nodeSpec.Cloud.Baremetal.Tinkerbell.OsImageUrl,
-		HardwareRef: nodeSpec.Cloud.Baremetal.Tinkerbell.HardwareRef,
+	tinkerbellConfig := nodeSpec.Cloud.Baremetal.Tinkerbell
+
+	// Validate required fields in Tinkerbell configuration
+	if tinkerbellConfig.HardwareRef.String() == "/" || tinkerbellConfig.OsImageUrl == "" {
```

Comment:
Would be nice if we can separate the `If` conditions out. For instance, if someone doesn't provide OsImageUrl but does provider HardwareRef, they'll still get an error stating "HardwareRef and OsImageUrl must be provided".